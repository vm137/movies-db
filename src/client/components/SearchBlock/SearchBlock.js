import React, {Component} from 'react';
import enterArrow from '../../img/enter-arrow.svg';
import './SearchBlock.scss';

export default class SearchBlock extends Component {

    constructor(props) {
        super(props);
        this.state = {
            movies: [],
            searchByButtons: true
        };
    }

    componentDidMount() {

        /* eslint-disable no-undef */
        fetch("http://react-cdp-api.herokuapp.com/movies")
            .then(function (response) {
                return response.json();
            })
            .then(function (myJson) {
                console.log(myJson);
            });
    }

    changeButtonsColor() {
        this.setState({searchByButtons: !this.state.searchByButtons});
    }

    render() {

        return (
          <div className="search-block-wrapper">
            <div className="search-block">
              <p className="find-title">find your movie</p>

              <div className="input-wrapper">
                <input className="search-input" type="text" />
                <img className="img-arrow" src={enterArrow} alt="enter" />
              </div>

              <div className="search-buttons">
                <span className="search-by">search by</span>
                <button
                  className={`btns-search btn-title ${this.state.searchByButtons ? "btn-bg-red" : "btn-bg-grey"}`}
                  onClick={this.changeButtonsColor.bind(this)}
                  type="button">title</button>
                <button
                  className={`btns-search btn-genre ${!this.state.searchByButtons ? "btn-bg-red" : "btn-bg-grey"}`}
                  onClick={this.changeButtonsColor.bind(this)}
                  type="button">genre</button>

                <button className="btns-search btn-search" type="submit">search</button>
              </div>

            </div>
          </div>
        )
    }

}
