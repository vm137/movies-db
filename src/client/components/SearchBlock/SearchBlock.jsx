import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchResults from '../SearchResults';
import Logo from '../Logo';
import Utils from '../Utils/Utils';
import COLOR from '../constants/constants';
import enterArrow from '../../img/enter-arrow.svg';
import './style.scss';

export default class SearchBlock extends Component {
  static propTypes = {
    numberFoundMovies: PropTypes.number.isRequired,
    movies: PropTypes.arrayOf(PropTypes.object).isRequired,
    searchCB: PropTypes.PropTypes.func.isRequired,
    onClick: PropTypes.PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      firstMount: true,
      titleButtonSelected: true,
      inputValue: 'you', // TODO: make '' after testing.
      sortByRelease: true,
    };
    this.textInput = React.createRef();
    this.onKeyPress = this.handleKeyPress.bind(this);
    this.makeSearch = this.makeSearch.bind(this);
    this.swapSearchByButtonsColor = this.swapSearchByButtonsColor.bind(this);
    this.handleSortByClick = this.handleSortByClick.bind(this);
  }


  componentDidMount() {
    this.textInput.focus();

    // TODO: delete after testing.
    this.makeSearch();
  }

  swapSearchByButtonsColor() {
    this.setState(prevState => ({ titleButtonSelected: !prevState.titleButtonSelected }));
  }

  makeSearch() {
    this.setState({ firstMount: false });
    const { titleButtonSelected, inputValue } = this.state;
    const { searchCB } = this.props;
    const searchBy = titleButtonSelected ? 'title' : 'genres';
    searchCB(inputValue, searchBy);
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value,
    });
  }

  handleSortByClick() {
    const { sortByRelease } = this.state;
    this.setState({
      sortByRelease: !sortByRelease,
    });
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      this.makeSearch();
    }
  }

  render() {
    const { numberFoundMovies, movies, onClick } = this.props;
    const {
      inputValue, firstMount, titleButtonSelected, sortByRelease,
    } = this.state;
    const sortReleaseColor = sortByRelease ? COLOR.NETFLIX_RED : COLOR.NETFLIX_GREY;
    const sortRatingColor = sortByRelease ? COLOR.NETFLIX_GREY : COLOR.NETFLIX_RED;

    return (
      <div className="search-block-wrapper">
        <div className="search-box-wrapper">
          <Logo />
          <div className="search-block">
            <p className="find-title">find your movie</p>

            <div className="input-wrapper">
              <input
                value={inputValue}
                onChange={evt => this.updateInputValue(evt)}
                onKeyPress={this.onKeyPress}
                ref={(input) => { this.textInput = input; }}
                className="search-input"
                type="text"
              />
              <img className="img-arrow" src={enterArrow} alt="enter" />
            </div>

            <div className="search-buttons">
              <span className="search-by">search by</span>
              <button
                className={`btns-search btn-title ${titleButtonSelected ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.swapSearchByButtonsColor}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
title
              </button>
              <button
                className={`btns-search btn-genre ${!titleButtonSelected ? 'btn-bg-red' : 'btn-bg-grey'}`}
                onClick={this.swapSearchByButtonsColor}
                onKeyPress={Utils.preventPressEnter}
                type="button"
              >
genre
              </button>

              <button
                onClick={this.makeSearch}
                className="btns-search btn-search"
                type="submit"
              >
search
              </button>
            </div>
          </div>
        </div>
        <div className="search-details">
          <div className="number-found-movies">
            {numberFoundMovies}
            {' '}
            movies found
          </div>
          <div className="sort-by">
            Sort by:
            <span
              role="button"
              tabIndex={0}
              onKeyPress={() => {}}
              className="sort-release-date"
              style={{ color: sortReleaseColor }}
              onClick={this.handleSortByClick}
            >
              release date
            </span>
            <span
              role="button"
              tabIndex={-1}
              onKeyPress={() => {}}
              className="sort-rating"
              style={{ color: sortRatingColor }}
              onClick={this.handleSortByClick}
            >
rating
            </span>
          </div>
        </div>
        {!firstMount && (
        <SearchResults onClick={onClick} movies={movies} />
        )}
      </div>
    );
  }
}
