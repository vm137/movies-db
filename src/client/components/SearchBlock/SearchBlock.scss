.search-box-wrapper {
    height: 200px;
    position: relative;

    &::after {
        content: "";
        background: url('../../img/old-cinema.png');
        opacity: 0.8;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        position: absolute;
        z-index: -1;
    }

    .search-block {
        position: absolute;
        top: 50px;
        left: 5%;
        width: 90%;

        .find-title {
            margin-top: 12px;
            color: #fff;
            font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
            font-size: 16px;
            text-transform: uppercase;
            text-shadow: 1px 1px 2px rgb(0, 0, 0);
        }

        .input-wrapper {
            position: relative;
            margin-top: 8px;

            .search-input {
                width: 100%;
                height: 28px;
                margin-top: 12px;
                color: #fff;
                padding-left: 12px;
                padding-bottom: 4px;
                border: none;
                border-bottom: 1px #ff594b solid;
                outline: none;
                font-size: 14px;
                box-sizing: border-box;
                background-color: #1d1d1d;
            }

            .img-arrow {
                position: absolute;
                width: 20px;
                top: 16px;
                right: 8px;
            }
        }

        .search-buttons {
            margin-top: 22px;

            .search-by {
                color: #fff;
                font-size: 12px;
                font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
                text-transform: uppercase;
            }
            .btns-search {
                border: 0;
                color: #fff;
                width: 90px;
                height: 26px;
                border-radius: 4px;
                outline: none;
                font-size: 12px;
                font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
                text-transform: uppercase;

                &:hover {
                    background-color: #ff866c;
                }
                &:active {
                    background-color: #151515;
                }
            }
            .btn-title {
                margin-left: 10px;
            }
            .btn-genre {
                margin-left: 16px;
            }
            .btn-search {
                float: right;
                background-color: #ff594b;
            }
        }
    }
}

.btn-bg-red {background-color: #ff594b !important;}
.btn-bg-grey {background-color: #808080 !important;}
