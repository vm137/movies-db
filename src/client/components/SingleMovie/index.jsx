import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import SingleMovie from './SingleMovie';
import { eraseSingleMovieAction, fetchSingleMovieAction } from '../../actions/actions';

const mapStateToProps = state => ({
  movieR: state.movie,
});

const mapDispatchProps = dispatch => ({
  eraseSingleMovieAction: () => dispatch(eraseSingleMovieAction()),
  fetchSingleMovie: movieId => dispatch(fetchSingleMovieAction(movieId)),
});

export default withRouter(connect(mapStateToProps, mapDispatchProps)(SingleMovie));
