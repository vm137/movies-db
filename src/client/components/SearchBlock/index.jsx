import { connect } from 'react-redux';
import SearchBlock from './SearchBlock';
import { fetchMoviesAction } from '../../actions/actions';

const mapStateToProps = state => ({
  moviesTotalR: state.total,
});

const mapDispatchProps = dispatch => ({
  fetchMovies: (searchString, searchBy) => dispatch(fetchMoviesAction(searchString, searchBy)),
});

export default connect(mapStateToProps, mapDispatchProps)(SearchBlock);
