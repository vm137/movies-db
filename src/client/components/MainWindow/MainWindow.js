import React, {Component} from 'react';
import SearchBlock from '../SearchBlock/SearchBlock';
import SingleMovie from '../SingleMovie/SingleMovie';

import './MainWindow.scss';
import fakeData from "../constants/fakeData";

export default class MainWindow extends Component {

    constructor(props) {
        super(props);
        this.state = {
            searchBlock: true
        };
    }

    componentWillMount() {
        this.makeFetch();
    }

    makeFetch() {
        /* eslint-disable no-undef */
        fetch("http://react-cdp-api.herokuapp.com/movies")
            .then(function (response) {
                return response.json();
            })
            .then(myJson => {
                    // console.log(myJson);
                    this.state.movies = myJson;
                    console.log(this.state.movies);
                }
            )
        //     .catch(
        //     (error) => {
        //         console.log("=== fetch unsuccessful.");
        //         console.log(error);
        //
        //         this.state.movies = fakeData;
        //     }
        // );
    }

    render() {
        return (
          <div>
            { this.state.searchBlock ? <SearchBlock /> : <SingleMovie /> }
          </div>
        );
    }
}
