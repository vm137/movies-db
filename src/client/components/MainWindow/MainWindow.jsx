import React, {Component} from 'react';
import SearchBlock from '../SearchBlock/SearchBlock';
import SingleMovie from '../SingleMovie/SingleMovie';
import './MainWindow.scss';


export default class MainWindow extends Component {

    constructor(props) {
        super(props);
        this.makeFetch = this.makeFetch.bind(this);
        this.state = {
            searchBlock: true,
            foundMovies: 0,
            movies: []
        };
    }

    makeFetch(searchString = '', searchBy = '', offset = '', limit = '') {
        let callString = 'http://react-cdp-api.herokuapp.com/movies';
        callString += `?search=${searchString}&searchBy=${searchBy}&offset=${offset}&limit=${limit}`;

        /* eslint-disable no-undef */
        fetch(callString)
            .then(function (response) {
                return response.json();
            })
            .then(myJson => {
                this.setState({
                    movies: myJson.data,
                    foundMovies: myJson.total
                })
            }
            )
        //     .catch(
        //     (error) => {
        //         console.log("=== fetch unsuccessful.");
        //         console.log(error);
        //
        //         this.state.movies = fakeData;
        //     }
        // );
    }

    render() {
        return (
          <div>
            { this.state.searchBlock ?
              <SearchBlock movies={this.state.movies} foundMovies={this.state.foundMovies} searchCB={this.makeFetch} /> :
              <SingleMovie movie="single movie 1" movies={this.state.movies} />
            }
          </div>
        );
    }
}
