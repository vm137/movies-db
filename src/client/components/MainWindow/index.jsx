import { connect } from 'react-redux';
import MainWindow from './MainWindow';

const mapStateToProps = state => ({
  showPage: state.showPage,
});

export default connect(mapStateToProps)(MainWindow);
