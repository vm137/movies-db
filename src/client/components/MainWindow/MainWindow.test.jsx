import { shallow } from 'enzyme';
import React from 'react';
import MainWindow from './MainWindow';
import SearchBlock from '../SearchBlock';
import SingleMovie from '../SingleMovie';

describe('<MainWindow />', () => {
  it('renders matching snapshot', () => {
    const wrapper = shallow(<MainWindow showPage="searchBlock" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with <SearchBlock />', () => {
    const wrapper = shallow(<MainWindow showPage="searchBlock" />);
    expect(wrapper.find(SearchBlock)).toHaveLength(1);
  });

  it('render with <SingleMovie />', () => {
    const wrapper = shallow(<MainWindow showPage="singleMovie" />);
    expect(wrapper.find(SingleMovie)).toHaveLength(1);
  });
});
