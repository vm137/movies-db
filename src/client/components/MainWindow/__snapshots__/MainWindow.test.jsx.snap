// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainWindow /> renders matching snapshot 1`] = `
<div
  className="mainWindow"
>
  <Connect(SearchBlock) />
</div>
`;
